### PR TITLE
Add device testing reference

### PR DESCRIPTION
**Pull request by doanac**
Signed-off-by: Andy Doan <andy@foundries.io>

### CODE REVIEW

**Comment by doanac** [comment]
Docs for e2c6db3db584084da3e918a30e795194324fdde2 are browesable at: https://ci.foundries.io/projects/fio-docs/builds/443/docs/artifacts/html/index.html

**Comment by doanac** [comment]
I've placed a "preview" version at https://docs.foundries.io/latest/reference/device-testing.html so that you can see how it renders.

File: source/reference/device-testing.rst [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+      "target": "raspberrypi3-64-lmp-8"
+    }
+    # The CreatedOn value will be auto-populated by the server.
+
+  **RETURNS**: A location to the new Test object::
+
+    HTTP 201
+    LOCATION: /tests/12345
+
+PUT /tests/
```
Comment (mike-sul) [inline source/reference/device-testing.rst:56]: Shouldn't it be some ID/test-name at the end?

File: source/reference/device-testing.rst [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+-------
+
+POST /tests/
+~~~~~~~~~~~~
+
+  Create a new test (``status=RUNNING``)::
+
+    # Content-type: application/json
+    {
+      "name": "test-name",
```
Comment (mike-sul) [inline source/reference/device-testing.rst:46]: So we can POST many tests with the same name, the Location returned from the post call includes this test run ID?

Comment (doanac) [reply source/reference/device-testing.rst:46]: correct. I often runt the "LTP" test many times

File: source/reference/device-testing.rst (hunk 1 above)
Comment (doanac) [reply source/reference/device-testing.rst:56]: yes. copy/paste mistake

Comment (doanac) [reply source/reference/device-testing.rst]: fixed with a git force push

**Comment by doanac** [comment]
Docs for 1d6b1588c9dd58edb304fbf33770cb7ce9fd7e44 are browesable at: https://ci.foundries.io/projects/fio-docs/builds/444/docs/artifacts/html/index.html

**Pull request merged**